src/roles/vsc-upgrade-deploy/tasks/main.yml

```diff
@@ -113,4 +113,15 @@
       that: "uptime_after_reboot.stdout < uptime_before_reboot.stdout"
       fail_msg: "System uptime after reboot step suggests there were issues during reboot. Check log for errors."
 
+  - name: Get primary config filename
+    set_fact:
+      vsc_config_filename: "{{ lookup('file', backup_folder + '/primary_config_filename') }}"
+
+  - name: Copy VSC config file to the VSC
+    delegate_to: localhost
+    shell: >-
+      sshpass -p'{{ vsc_password | default(vsc_default_password) }}' scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null
+      {{ proxy_conf | default('') }} {{ backup_folder }}/{{ vsc_config_filename }} {{ vsc_username | default(vsc_default_username) }}@{{ mgmt_ip }}:
+    no_log: "{{ lookup('env', 'METROAE_NO_LOG') or 'true' }}"
```

Comment:
what if the VSC backup is not run and VSC upgrade is directly run. Should we have a when condition indicating copy the config only if vsc_config_filename is present. I do not know what lookup file returns if not present. None or otherwise. Need to try it out.

Reply:
Good catch. I guess users could skip the backup if they want to. @smrafey please run an experiment to see what happens if we try to use lookup with a file path that doesn't exist. We'll need a when condition on the copy part to make sure the file is found